### PR TITLE
[doc] TimeVal

### PR DESCRIPTION
**Pull request by Frenzie**

### CODE REVIEW

File: frontend/ui/timeval.lua [hunk 1]
```diff
@@ -1,10 +1,43 @@
+--[[--
+A simple module to deal with time values.
+
+@usage
+    local TimeVal = require("ui/timeval")
+
+    local tv_start = TimeVal:now()
+    -- Do some stuff.
+    -- You can add and substract `TimeValue` objects.
```
Comment (poire-z) [inline frontend/ui/timeval.lua]: TimeVal, not TimeVal**ue**.
(idem below when documenting TimeVal:now())

File: frontend/ui/timeval.lua [hunk 2]
```diff
@@ -1,10 +1,43 @@
...
+    local tv_duration = TimeVal:now() - tv_start
+    -- If you need more precision (like 2.5 s),
+    -- you can add the milliseconds to the seconds.
+    local tv_duration = tv_duration.sec + tv_duration.usec/1000000
```
Comment (poire-z) [inline frontend/ui/timeval.lua]: A bit confusing: your first tv_duration is still a TimeVal object (which can be used for other TimeVal arithmetic).
The second is a number (whether you add or nor the `.usec` if you need more precision).
(idem below when documenting TimeVal:now())

File: frontend/ui/timeval.lua [hunk 3]
```diff
@@ -1,10 +1,43 @@
+--[[--
+A simple module to deal with time values.
```
Comment (poire-z) [inline frontend/ui/timeval.lua]: A simple module to compare and do arithmetic with time/timedelta values. (?)

File: frontend/ui/timeval.lua (hunk 2 above)
Comment (Frenzie) [reply frontend/ui/timeval.lua]: The confusing example comes straight from your code, lol :-P

Comment (poire-z) [reply frontend/ui/timeval.lua]: :) ok, but if i'm dirty changing a variable type just to not introduce an additional local variable (i often do that), that should not be documented :)

**Pull request merged**